### PR TITLE
Wip cppcheck errors

### PR DESCRIPTION
**Pull request by badone**
Various fixes for the cppcheck errors seen http://people.redhat.com/bhubbard/cppcheck/cppcheck.17-04-05.txt

### CODE REVIEW

File: src/test/librados/io.cc [hunk 1]
```diff
@@ -826,6 +827,12 @@ TEST_F(LibRadosIoEC, OverlappingWriteRoundTrip) {
   char *buf = (char *)new char[dbsize];
   char *buf2 = (char *)new char[bsize];
   char *buf3 = (char *)new char[dbsize];
+  auto lam = [&] {
+            delete[] buf;
+            delete[] buf2;
+            delete[] buf3;
+          };
+  scope_guard<decltype(lam)> sg(std::forward<decltype(lam)>(lam));
```
Comment (tchaikov) [inline src/test/librados/io.cc]: why not use `make_scope_guard()`?

Comment (tchaikov) [inline src/test/librados/io.cc]: why not use `make_scope_guard()`?

Comment (badone) [inline src/test/librados/io.cc]: Yes, good question. The reason is that cppcheck issues an error about a memory leak (false positive) when make_scope_guard is used but does not when creating a scope_guard object with a named lambda. I have reported this upstream and am trying to get access to create a cppcheck issue.

File: src/test/librados/io.cc [hunk 2]
```diff
@@ -826,6 +827,12 @@ TEST_F(LibRadosIoEC, OverlappingWriteRoundTrip) {
   char *buf = (char *)new char[dbsize];
   char *buf2 = (char *)new char[bsize];
   char *buf3 = (char *)new char[dbsize];
+  auto lam = [&] {
```
Comment (badone) [inline src/test/librados/io.cc]: Well, yes, we could change everything to use the stack, that would be easy, but I figured I would follow the lead set by the original author.

**Comment by badone** [comment]
`Error EIO: load dlopen(/home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libec_FAKE.so): /home/jenkins-build/build/workspace/ceph-pull-requests/build/lib/libec_FAKE.so: cannot open shared object file: No such file or directory`

Checking if the above is transient.

Jenkins test this please.

File: src/test/librados/io.cc (hunk 2 above)
Comment (tchaikov) [inline src/test/librados/io.cc]: i was wrong, C++ (at this moment) does not support VLA. but GCC does.

File: src/test/librados/io.cc (hunk 1 above)
Comment (tchaikov) [inline src/test/librados/io.cc]: but i don't think we should trade the readability for appeasing the static analyzer if the code is correct. moreover, we can avoid use names like `lam` (i figured out it's the short for lambda just now).

File: src/test/librados/io.cc [hunk 3]
```diff
@@ -918,13 +918,13 @@ TEST_F(LibRadosIoEC, AppendRoundTrip) {
...
-            delete[] unalignedbuf;
-          };
-  scope_guard<decltype(lam)> sg(std::forward<decltype(lam)>(lam));
+  auto cleanup = [&] {
+    delete[] buf;
+    delete[] buf2;
+    delete[] buf3;
+    delete[] unalignedbuf;
+  };
+  scope_guard<decltype(cleanup)> sg(std::forward<decltype(cleanup)>(cleanup));
```
Comment (tchaikov) [inline src/test/librados/io.cc]: nit, `sg(std::move(cleanup))` would suffice. we know the type of cleanup here. and we do want to rvalue reference here.

Comment (badone) [inline src/test/librados/io.cc]: Right, as always, thanks!

**Pull request merged**